discovery/dns/dns.go

```diff
@@ -317,7 +317,7 @@ func askServerForName(name string, queryType uint16, client *dns.Client, servAdd
 	}
 
 	response, _, err := client.Exchange(msg, servAddr)
-	if err == dns.ErrTruncated {
+	if err != nil || response.Truncated {
```

Comment:
```suggestion
	if response.Truncated {
```

the `err != nil` check is being done in L328. If an error other than truncated message is thrown, then that error's details will never be shown with this conditional block.

Reply:
Pinging @simonpasquier as he LGTM'ed this. Please let me know if this review comment holds.

Reply:
Sure! I am editing my comment to remove the GH suggestion.

Reply:
I'm not seeing where errid is needed or where it comes from, but I've made the rest of the changes; committing now.

Reply:
Ah I see now. I will resolve it. Thanks.